### PR TITLE
CAD-2572 Direct submission of the ChainSync headers served metric to EKG

### PR DESCRIPTION
**Pull request by jutaro**
1. Bump `iohk-monitoring-framework` to the latest `master`, that includes EKG server extraction.
2. Define `EKGDirect` -- a direct handle onto the EKG server/store used by the EKG backend of the logging framework.
3. Stop submitting `cardano.node.metrics.served.header.counter.int` to the switchboard.
4. Submit it via `EKGDirect`, instead.

### CODE REVIEW

**Review by deepfire — APPROVED** [review]
LGTM, thank you Jurgen!

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 1]
```diff
@@ -420,6 +430,28 @@ traceD tr meta msg d = traceNamedObject tr (meta, LogValue msg (PureD d))
...
+ case SMap.lookup name registeredMap of
+   Just gauge -> Gauge.set gauge (fromIntegral val)
+   Nothing -> do
+     gauge <- EKG.getGauge name (ekgServer ekgDirect)
+     let registeredMap' = SMap.insert name gauge registeredMap
+     writeIORef (ekgGauges ekgDirect) registeredMap'
+     Gauge.set gauge (fromIntegral val)
+
+_sendEKGDirectDouble :: EKGDirect -> Text -> Double -> IO ()
+_sendEKGDirectDouble ekgDirect name val = do
```
Comment (newhoggy) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:445]: Why does this function start with an underscore?

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 2]
```diff
@@ -420,6 +430,28 @@ traceD tr meta msg d = traceNamedObject tr (meta, LogValue msg (PureD d))
...
+sendEKGDirectInt :: Integral a => EKGDirect -> Text -> a -> IO ()
+sendEKGDirectInt ekgDirect name val = do
+ registeredMap <- readIORef (ekgGauges ekgDirect)
+ case SMap.lookup name registeredMap of
+   Just gauge -> Gauge.set gauge (fromIntegral val)
+   Nothing -> do
+     gauge <- EKG.getGauge name (ekgServer ekgDirect)
+     let registeredMap' = SMap.insert name gauge registeredMap
+     writeIORef (ekgGauges ekgDirect) registeredMap'
+     Gauge.set gauge (fromIntegral val)
```
Comment (newhoggy) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:442]: What happens if this is run from multiple threads and `getGauge` gets called multiple times for the same name?

File: cardano-node/src/Cardano/Tracing/Tracers.hs [hunk 3]
```diff
@@ -544,6 +572,13 @@ mkConsensusTracers trSel verb tr nodeKern fStats = do
        <*> counting (liftCounting staticMeta name "slot-is-immutable" tr)
        <*> counting (liftCounting staticMeta name "node-is-leader" tr)
 
+   traceServedCount :: Maybe EKGDirect -> STM.TVar Int -> TraceChainSyncServerEvent blk -> IO ()
+   traceServedCount Nothing _ _ = pure ()
+   traceServedCount (Just ekgDirect) tHeadersServed ev =
+     when (isRollForward ev) $ do
+       count <- STM.modifyReadTVarIO tHeadersServed (+1)
```
Comment (newhoggy) [inline cardano-node/src/Cardano/Tracing/Tracers.hs:579]: Are we still using TVars?  I had the impression that we were going to replace the use of TVars.

File: cardano-node/src/Cardano/Tracing/Tracers.hs (hunk 1 above)
Comment (deepfire) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:445]: @newhoggy, this is currently unused, but will be immediately used by anyone who needs to post a `Double` metric.

We don't want then to be running around for recipes.. : -)

File: cardano-node/src/Cardano/Tracing/Tracers.hs (hunk 3 above)
Comment (newhoggy) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:579]: Mostly due to this comment https://input-output-rnd.slack.com/archives/GR599HMFX/p1613382544290700

Comment (deepfire) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:579]: @newhoggy, can you tell a bit more -- what should we be using instead?  I'm afraid I'm not up-to-date..

Comment (newhoggy) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:579]: Found the comment https://github.com/input-output-hk/cardano-node/pull/2366#discussion_r576062283

Comment (newhoggy) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:579]: I added @coot to seek clarification on this question.

Comment (deepfire) [reply cardano-node/src/Cardano/Tracing/Tracers.hs:579]: Good point, we should be using https://hackage.haskell.org/package/ekg-core-0.1.1.7/docs/System-Metrics-Counter.html

**Comment by newhoggy** [comment]
💯

**Review by newhoggy — APPROVED** [review]

**Pull request merged**